capstone/src/main/java/com/google/sps/servlets/YoutubeServlet.java

```diff
@@ -38,20 +38,27 @@
   private static final String KEY = "API_KEY";
   private static final String NUM_RESULTS = "5";
   private static final MediaType JSON = MediaType.get("application/json; charset=utf-8");
+  private String postRequestBodyData;
+  private String channelId;
+  private String videoId; 
   OkHttpClient client = new OkHttpClient();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String ChannelId = request.getParameter("channelId");
-    String videoId = request.getParameter("videoId");
-    String completeUrl = (ChannelId != null) ? 
-      (BASE_URL + "&allThreadsRelatedToChannelId=" + ChannelId + "&maxResults=" + NUM_RESULTS + "&key=" + KEY) : 
-        (BASE_URL + "&videoId=" + videoId + "&maxResults=" + NUM_RESULTS + "&key=" + KEY);
+    String completeUrl = (postRequestBodyData.length() == 24) ? 
+      (BASE_URL + "&allThreadsRelatedToChannelId=" + postRequestBodyData + "&maxResults=" + NUM_RESULTS + "&key=" + KEY) : 
+        (BASE_URL + "&videoId=" + postRequestBodyData + "&maxResults=" + NUM_RESULTS + "&key=" + KEY);
     String output = get(completeUrl);
     response.setContentType("application/json");
     response.getWriter().println(output);  
   }
   
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    postRequestBodyData = request.getReader().readLine().replaceAll("^\"+|\"+$", "");
```

Comment:
nit: Could we put a comment for what this regular expression represents?

Reply:
I have chosen to use Java's inbuilt ".trim()" function instead of a regular expression in "address nits and add youtube caller support"